libr/core/cmd_debug.c

```diff
@@ -2779,12 +2705,7 @@ static int cmd_debug_step (RCore *core, const char *input) {
 	switch (input[1]) {
 	case 0:
 		r_reg_arena_swap (core->dbg->reg, true);
-		r_debug_reg_sync (core->dbg, R_REG_TYPE_GPR, false);
 		r_debug_step (core->dbg, times);
-		if (checkbpcallback (core)) {
```

Comment:
iirc regsync is done before step because BSD requires to pass the updated PC in the PT_STEP/PT_CONT calls